### DISCUSSION TITLE
Make `navigate` a top-level export or available via simpler hook

### DISCUSSION THREAD

**Discussion by MarkLyck** [TanStack/router#384] [Ideas]
I have a bunch of components that don't really need any router "state". But they do need to programmatically navigate to another route.

Having to import both `useMatch` from the router and the `route` so I can do `const { navigate } = useMatch(myRoute.id)` is a bit much.

Ideally it would be great if I can just import navigate directly from the package `import { navigate } from '@tanstack/react-router'`. But I can see how that won't work for cases where there might be multiple routers.

As a second option I would like to have a simpler `useNavigate()` hook that works the same as react-router-dom.

`import { useNavigate } from '@tanstack/react-router`
`const navigate = useNavigate()`

**Comment by ryanstaniforth** [comment]
You should only need to import your router instance. You can then use `router.navigate('/your/route');`. In this example, `'/your/route'` is type safe, just make sure you include the following where you define your routes:

```typescript
declare module '@tanstack/react-router' {
  interface RegisterRouter {
    router: typeof router;
  }
}
```

Does this help?

**Comment by tannerlinsley** [comment]
Also remember that any global or contextual router import will not have type safety for relative navigations. Which is why getting the function off of the match is nice. It has the relative type safety.

Tanner Linsley
